doc/api/icu.md

```diff
@@ -0,0 +1,76 @@
+# icu
+
+    Stability: 1 - Experimental
+
+The `icu` module provides an interface to Unicode and Internationalization
+functionality provided by the ICU4C library.
+
+```js
+const icu = require('icu');
+```
+
+## icu.transcode(buf, from_enc, to_enc)
+
+* `buf` {Buffer} A `Buffer` instance
+* `from_enc` {string} The current encoding
+* `to_enc` {string} The target encoding
+
+Re-encodes the given `Buffer` from one character encoding to another. Returns
+a new `Buffer` instance.
```

Comment:
Potentially. The key issue with this is that the returned Buffer could potentially be quite a bit larger than the source so we would have to decide the acceptable behavior if the passed in Buffer is not large enough. Would we throw or truncate?